tools/plugin/alsaplug/ctl.c

```diff
@@ -618,22 +541,39 @@ SND_CTL_PLUGIN_DEFINE_FUNC(sof)
 	plug->module_prv = ctl;
 
 	/* parse the ALSA configuration file for sof plugin */
-	err = plug_parse_conf(plug, name, root, conf);
+	err = plug_parse_conf(plug, name, root, conf, true);
 	if (err < 0) {
 		SNDERR("failed to parse config: %s", strerror(err));
 		goto error;
 	}
 
-	/* create message queue for IPC */
-	err = plug_mq_init(&ctl->ipc, plug->tplg_file, "ipc", 0);
-	if (err < 0)
+	/* init IPC tx message queue name */
+	err = plug_mq_init(&ctl->ipc_tx, "sof", "ipc-tx", 0);
+	if (err < 0) {
+		SNDERR("error: invalid name for IPC tx mq %s\n", plug->tplg_file);
+		goto error;
+	}
+
+	/* open the sof-pipe IPC tx message queue */
+	err = plug_mq_open(&ctl->ipc_tx);
+	if (err < 0) {
+		SNDERR("error: failed to open sof-pipe IPC mq %s: %s",
+		       ctl->ipc_tx.queue_name, strerror(err));
 		goto error;
+	}
+
+	/* init IPC rx message queue name */
+	err = plug_mq_init(&ctl->ipc_rx, "sof", "ipc-rx", 0);
+	if (err < 0) {
+		SNDERR("error: invalid name for IPC rx mq %s\n", plug->tplg_file);
+		goto error;
+	}
```

Comment:
This is a bit surprising to see the msg queue implementation change as part of this commit -- so to support volume controls, you need to change the IPC machinery quite a bit.

Reply:
@kv2019i the msg queue implementation has not been changed in this commit. The original implementation never worked. So it has been fixed to work here

Reply:
Ack @ranj063 . Info like this is important to catch in the git commit messages as they are the permenant history and vital info anyone who needs to maintain this code later on. I see you already update the commit, so good with me.